bindings/go/runtime/type.go

```diff
@@ -0,0 +1,166 @@
+package runtime
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// Typed is any object that is defined by a type that is versioned.
+type Typed interface {
+	// GetType returns the object's type
+	GetType() Type
+}
+
+// Type represents a structured type with an optional group, an optional version and a name.
+// It is used to identify the type of an object in a versioned API.
+// A Group is a namespace for related types,
+// a Version is a specific iteration of the type,
+// and Name is the name of the type.
+type Type struct {
+	Group   string
+	Version string
+	Name    string
+}
+
+// NewUngroupedVersionedType creates a new Type instance without a group and with a version.
+func NewUngroupedVersionedType(name, version string) Type {
+	return Type{Name: name, Version: version}
+}
+
+// NewUngroupedUnversionedType creates a new Type instance without a group and without a version.
+func NewUngroupedUnversionedType(name string) Type {
+	return Type{Name: name}
+}
+
+// NewType creates a new Type instance with a group and version.
+func NewType(group, version, name string) Type {
+	return Type{Group: group, Version: version, Name: name}
+}
+
+// TypeFromString parses a type string in the formats:
+// - "name" (unversioned, no group)
+// - "name/version" (versioned, no group)
+// - "group.name" (unversioned, with group)
+// - "group.name/version" (versioned, with group)
+func TypeFromString(typ string) (Type, error) {
+	parts := strings.Split(typ, "/")
+
+	// Only allow one or two parts (name or name/version)
+	if len(parts) > 2 {
+		return Type{}, fmt.Errorf("invalid type %q, too many segments", typ)
+	}
+
+	var namePart, versionPart string
+	if len(parts) == 1 {
+		namePart = parts[0] // Unversioned format
+	} else {
+		namePart, versionPart = parts[0], parts[1] // Versioned format
+	}
+
+	// Split name part to extract group (if present)
+	nameParts := strings.Split(namePart, ".")
+	if len(nameParts) < 1 {
```

Comment:
split always returns a element, even if there are no elements, it returns one empty element.

Reply:
mind telling me a case which I can test against that fails? I thought I covered most cases X)

Reply:
Well, you could say if it's `len(nameParts) == 1 && nameParts[0] == ""` if that happens then there is definitely a problem. :)

Reply:
Oh you mean an actual case. :D

To trigger the above, you are missing a testcase like this: `/v1`. If someone forgets to write the name part. :)

This would result here in an empty namePart but with the length of `1`.

Reply:
Ah that case is actually covered by checking if the Name is filled (line 80). I added a test case to make sure.

Reply:
Ah good catch, okay dokay.